codedigger/atcoder/tests/test_upsolve.py

```diff
@@ -0,0 +1,21 @@
+from django import test
+from .test_setup import TestSetUp
+from django.urls import reverse
+
+
+class TestUpsolve(TestSetUp):
+    def test_atcoder_upsolve(self):
+        test_url = reverse('at-upsolve') + '?practice=true'
+        token = self.login(self.client, self.login_url, self.user_data)
+        client = self.get_authenticated_client(token)
+        res = client.get(test_url, format="json")
+        self.assertEqual(len(res.data['result'][0]['problems']), 5)
```

Comment:
Make sure this data didn't get update if this user solved more contest. That means, problems of contest may not be fixed to 5. Please add other conditions in test.